### PR TITLE
[JIT] Fix torch.tensor for empty multidimensional-typed lists

### PR DESCRIPTION
**Pull request by eellison**
We were hitting an assert error when you passed in an empty `List[List[int]]` - this fixes that error by not recursing into 0-element tensors.

### CODE REVIEW

**Comment by SplitInfinity** [comment]
Out of curiosity, is this the assert in question: https://fburl.com/diffusion/am1ysuge

**Comment by eellison** [comment]
It was this assert failure: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/runtime/register_special_ops.cpp#L181 bc `ndim = 1`, `dim=0`, so we are expecting to find the list containing the actual data (IntList/BoolList/DoubleList), but instead we find a `int[][]`.

**Review by SplitInfinity — APPROVED** [review]
